File: AoC-2020/advent2/advent2.py
import re

valid=valxor=0

def get_values(file):
    while True:
        line = re.split("-| |: ",file.readline().strip())
        if len(line) != 4:
            break

        min, max, char, pwd = line  # unpack list
        check(int(min), int(max), char, pwd)

    print("Pt1: ", valid, "\nPt2: ", valxor)

def check(min, max, char, pwd):
    c=xor=0; global valid; global valxor
    for i in pwd:
        if i is char:
            c+=1

        if (pwd[min-1] is char) ^ (pwd[max-1] is char):
            xor = 1

    if c >= min and c <= max:
        valid+=1
    if xor == 1:
        valxor += 1;

file = open('input')
get_values(file)

File: AoC-2020/advent2/advent2.c
#include <stdio.h>

#define  MAX  1000

int check(int, int, char, char *);
int check2(int, int, char, char *);

int sum, valid, validxor;

int main(){
     int min, max, sum, i, j;
     char chara, c;
     char *pwd;
     valid = validxor = sum = 0;

     for (i = 0; i < MAX; i++){
          scanf("%d-%d %c: %s", &min, &max, &chara, pwd);

          check2(min, max, chara, pwd);
          check(min, max, chara, pwd);
     }
     printf("[!] Valid Pwds: %d\n", valid);
     printf("[!] Valid XOR Pwds: %d\n", validxor);
}

int check(int min, int max, char chara, char *pwd){
     int count, i;
     count = 0;
     for (i = 0; pwd[i] != '\0'; i++)
          if (pwd[i] == chara)
               count++;
     printf("%d %d\n", i, count);
     if (count >= min && count <= max)
          valid++;
}

int check2(int min, int max, char chara, char *pwd){
     int i, xor;
     xor = 0;
     for (i = 0; pwd[i] != '\0'; i++)
          if ((pwd[min-1] == chara && pwd[max-1] != chara) || (pwd[max-1] == chara && pwd[min-1] != chara))
               xor = 1;
          (xor == 1) ? validxor++ : 0 ;
     printf("%d[%c]:%d[%c]:%c:%s\n", min, pwd[min-1], max, pwd[max-1], chara, pwd);
     printf("XOR = %d / XOR++ = %d\n", xor, validxor);
}
